### PR TITLE
monaco editor integration

### PR DESCRIPTION
**Pull request by SYBIOTE**
version one of monaco intergration 
the monaco editor is connected to the file

needs some extensive QA to account for interactions with the file browser

### CODE REVIEW

File: packages/server-core/src/media/file-browser/file-browser.class.ts [hunk 1]
```diff
@@ -178,6 +178,8 @@ export class FileBrowserService
    */
   async create(directory: string, params?: FileBrowserParams) {
     const storageProvider = getStorageProvider(params?.query?.storageProviderName)
+
+    console.log('DEBUG directory', directory)
```
Comment (aditya-mitra) [inline packages/server-core/src/media/file-browser/file-browser.class.ts]: remove console log

File: packages/ui/src/components/editor/panels/Script/container/index.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+      )}
+    </AutoSizer>
+  )
+}
+
+export const ScriptPanel = () => {
+  return (
+    <>
+      <div className="flex h-full w-full flex-col">
+        <div className="flex h-full w-full flex-col">
```
Comment (aditya-mitra) [inline packages/ui/src/components/editor/panels/Script/container/index.tsx]: duplicate divs

File: packages/ui/src/components/editor/panels/Script/container/index.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    if (!scriptComponent?.src.value) return
+    if (!validateScriptUrl(entity, scriptComponent?.src.value)) return
+    clearErrors(entity, ScriptComponent)
+    updateScriptFile(getFileName(scriptComponent!.src.value), code)
+  }, [code])
+
+  const addScript = () => EditorControlFunctions.addOrRemoveComponent([entity], ScriptComponent, true)
+  useQuery([ScriptComponent])
+  return (
+    <AutoSizer>
```
Comment (aditya-mitra) [inline packages/ui/src/components/editor/panels/Script/container/index.tsx]: can we remove autosizer here? was planning on removing autosizer altogether from codebase at a later point

**Review by aditya-mitra — CHANGES_REQUESTED** [review]

File: packages/ui/src/components/editor/panels/Script/container/index.tsx (hunk 3 above)
Comment (SYBIOTE) [reply packages/ui/src/components/editor/panels/Script/container/index.tsx]: sure

File: packages/server-core/src/media/file-browser/file-browser.class.ts (hunk 1 above)
Comment (SYBIOTE) [reply packages/server-core/src/media/file-browser/file-browser.class.ts]: nice catch

File: packages/ui/src/components/editor/panels/Script/container/index.tsx (hunk 2 above)
Comment (SYBIOTE) [reply packages/ui/src/components/editor/panels/Script/container/index.tsx]: cool

**Review by SamMazerIR — APPROVED** [review]

**Pull request merged**